drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,644 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nordic Semiconductor nRF52 UARTE
+ */
+
+
+#include <uart.h>
+#include <gpio.h>
+#include <hal/nrf_uarte.h>
+
+/* Device data structure */
+typedef struct {
+	u8_t rx_data;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_t cb;	/**< Callback function pointer */
+	u8_t tx_buffer[CONFIG_UART_NRFX_RAM_BUFFER_SIZE];
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+} uarte_nrfx_data_t;
+
+typedef struct
+{
+	NRF_UARTE_Type * p_reg;        /* Pointer to registers structure */
+	uint8_t          drv_inst_idx; /* Driver instance index */
+} uarte_nrfx_instance_t;
+
+/**
+ * @brief Structure for UARTE configuration.
+ */
+typedef struct
+{
+	size_t max_chunk_len;		/* max allowed EasyDMA buffer size */
+	u32_t  pseltxd;			/* TXD pin number */
+	u32_t  pselrxd;			/* RXD pin number. */
+	u32_t  pselcts;			/* CTS pin number. */
+	u32_t  pselrts;			/* RTS pin number. */
+
+	uarte_nrfx_instance_t uarte;	/* Instance address and number. */
+	nrf_uarte_hwfc_t      hwfc;	/* Flow control configuration. */
+	nrf_uarte_parity_t    parity;	/* Parity configuration. */
+	nrf_uarte_baudrate_t  baudrate;	/* Baudrate. */
```

Comment:
What is the point in storing all these parameters that you only use at initialization?
Actually, you only need `max_chunk_len` and `uarte.p_reg`.